CMakeLists.txt

```diff
@@ -183,6 +183,10 @@ if(WITH_BLUESTORE)
 endif()
 
 include(CMakeDependentOption)
+CMAKE_DEPENDENT_OPTION(WITH_LIBURING "Build with liburing library support" OFF
```

Comment:
@rouming i take the liberty to use `CMAKE_DEPENDENT_OPTION` to disable io-uring support if WITH_BLUESTORE or HAVE_LIBAIO is disabled.